MSMAll/MSMAllPipeline.sh

```diff
@@ -216,31 +327,43 @@ log_Msg "fMRIProcSTRING: ${fMRIProcSTRING}"
 
 # run MSMAll
 "${HCPPIPEDIR}"/MSMAll/scripts/"${ModuleName}" \
-	--path="${StudyFolder}" \
-	--subject="${Subject}" \
-	--high-res-mesh="${HighResMesh}" \
-	--low-res-mesh="${LowResMesh}" \
-	--output-fmri-name="${OutputfMRIName}" \
-	--fmri-proc-string="${fMRIProcSTRING}" \
-	--input-pca-registration-name="${InputRegName}" \
-	--input-registration-name="${InputRegName}" \
-	--registration-name-stem="${OutputRegName}" \
-	--rsn-target-file="${RSNTemplates}" \
-	--rsn-cost-weights="${RSNWeights}" \
-	--myelin-target-file="${MyelinTarget}" \
-	--topography-roi-file="${TopographyROIs}" \
-	--topography-target-file="${TopographyTarget}" \
-	--iterations="${IterationModes}" \
-	--method="${Method}" \
-	--use-migp="${UseMIGP}" \
-	--ica-dim="${ICAdim}" \
-	--regression-params="${LowsICADims}" \
-	--vn="${VN}" \
-	--rerun="${ReRunIfExists}" \
-	--reg-conf="${RegConfPath}" \
-	--reg-conf-vars="${RegConfOverrideVars}" \
-	--msm-all-templates="${MSMAllTemplates}" \
-	--use-ind-mean="${UseIndMean}" \
-	--matlab-run-mode="${MatlabRunMode}"
-	
+    --path="${StudyFolder}" \
+    --session="${Session}" \
+    --high-res-mesh="${HighResMesh}" \
+    --low-res-mesh="${LowResMesh}" \
+    --output-fmri-name="${OutputfMRIName}" \
+    --fmri-proc-string="${fMRIProcSTRING}" \
+    --input-pca-registration-name="${InputRegName}" \
+    --input-registration-name="${InputRegName}" \
+    --registration-name-stem="${OutputRegName}" \
+    --rsn-target-file="${RSNTemplates}" \
+    --rsn-cost-weights="${RSNWeights}" \
+    --myelin-target-file="${MyelinTarget}" \
+    --topography-roi-file="${TopographyROIs}" \
+    --topography-target-file="${TopographyTarget}" \
+    --iterations="${IterationModes}" \
+    --method="${Method}" \
+    --use-migp="${UseMIGP}" \
+    --ica-dim="${ICAdim}" \
+    --regression-params="${LowsICADims}" \
+    --vn="${VN}" \
+    --rerun="${ReRunIfExists}" \
+    --reg-conf="${RegConfPath}" \
+    --reg-conf-vars="${RegConfOverrideVars}" \
+    --msm-all-templates="${MSMAllTemplates}" \
+    --use-ind-mean="${UseIndMean}" \
+    --matlab-run-mode="${MatlabRunMode}"
+
+#copy the registration result sphere from template back to timepoints.    
+if (( IsLongitudinal )); then
+    
+    for tp in ${SessionsLong[@]}; do
+        cp $StudyFolder/$TemplateSession/MNINonLinear/Native/$TemplateSession.SphericalDistortion.native.dscalar.nii \
+            $StudyFolder/$SessionLong/MNINonLinear/Native/$SessionLong.SphericalDistortion.native.dscalar.nii
```

Comment:
Why is this file being copied?  Is the spherical inflation distortion truly the same across timepoints?  It is possible that it isn't.

Reply:
I guess I am a bit confused here.  Where is this map normally generated from in cross-sectional?  I am not sure that the spherical inflation is exactly the same across the longitudinal timepoints (which is what would be implied by copying this file).

Reply:
Measuring the difference between a timepoint's midthickness and the single longitudinal native sphere can/should produce different results than the longitudinal template midthickness.  On the other hand, since the surface wasn't re-inflated to a sphere for longitudinal surfaces, it can be expected to contain some anatomical changes on top of the actual distortion seen by the spherical inflation algorithm.

Reply:
I don't think we know, but we can ask.